packages/common/src/internal-non-workflow/codec-types.ts

```diff
@@ -23,12 +23,14 @@ export type ReplaceNested<T, ToReplace, ReplaceWith> = T extends (...args: any[]
   ? T
   : [keyof T] extends [never]
     ? T
-    : T extends { [k: string]: ToReplace }
-      ? {
-          [P in keyof T]: ReplaceNested<T[P], ToReplace, ReplaceWith>;
-        }
-      : T extends ToReplace
-        ? ReplaceWith | Exclude<T, ToReplace>
-        : {
+    : T extends Record<string, string> // Special exception for nexusHeader.
```

Comment:
I vaguely remember struggling with this mapped type missing some transformations. This is probably the reason Roey had to modify it here, but I'm worried it may still be in some other aspects.

@THardy98 Can you please get a look at the [Type Fest](https://github.com/sindresorhus/type-fest) project; they probably have some type definition that does the same as this (name may differ). If they do, please copy that type definition and its dependencies into common/src/type-helpers.ts, and get rid of this one.

Reply:
Took a look, the closest match is likely [MergeDeep](https://github.com/sindresorhus/type-fest/blob/ed8c987129fe180021348d8fdbd52e662ffd69a1/source/merge-deep.d.ts#L58), but I don't think it's exactly what we're looking for.